src/Form/AuthenticationSettingsForm.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_authentication\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Settings form for module.
+ */
+class AuthenticationSettingsForm extends ConfigFormBase {
+
+  /**
+   * Name of the config being edited.
+   */
+  const CONFIGNAME = 'oe_authentication.settings';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'oe_authentication_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['protocol'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Application authentication protocol'),
+      '#default_value' => $this->config(static::CONFIGNAME)->get('protocol'),
+    ];
+    $form['register_path'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Application register path'),
+      '#default_value' => $this->config(static::CONFIGNAME)->get('register_path'),
+    ];
+    $form['validation_path'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Application validation path'),
+      '#default_value' => $this->config(static::CONFIGNAME)->get('validation_path'),
+    ];
+    $form['assurance_level'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Application assurance levels'),
+      '#default_value' => $this->config(static::CONFIGNAME)->get('assurance_level'),
+    ];
+    $form['ticket_types'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Application available ticket types'),
+      '#default_value' => $this->config(static::CONFIGNAME)->get('ticket_types'),
+    ];
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->config(static::CONFIGNAME)
+      ->set('protocol', $form_state->getValues()['protocol'])
```

Comment:
@nagyad FormStateInterface has a `getValue()` method that allows to ask for a single submitted value. Either use that or use `getValues()` into an array variable at the top and access its keys here. Preferably the first one.